### PR TITLE
updated pymaker, use pymaker requirements, improve node support

### PR DESCRIPTION
**Pull request by EdNoepel**
Not really familiar with this bot; please review and test.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,7 +117,7 @@ In order to clone the project and install required third-party packages please e
 git clone https://github.com/makerdao/arbitrage-keeper.git
 cd arbitrage-keeper
 git submodule update --init --recursive
-pip3 install -r requirements.txt
+pip3 install -r lib/pymaker/requirements.txt
```
Comment (EdNoepel) [inline README.md:120]: Maintaining it's own requirements seemed to be a source of code rot.  I didn't see anything special outside of what `pymaker` requires itself.

File: simple_arbitrage_keeper/simple_arbitrage_keeper.py [hunk 2]
```diff
@@ -28,7 +28,7 @@
 from simple_arbitrage_keeper.oasis_api import OasisAPI
 from simple_arbitrage_keeper.transfer_formatter import TransferFormatter
 
-from pymaker import Address
+from pymaker import Address, web3_via_http
```
Comment (EdNoepel) [inline simple_arbitrage_keeper/simple_arbitrage_keeper.py:31]: This should help the community use SSL-based node providers.

**Comment by KentonPrescott** [comment]
Added a warning (Keeper doesn't exclusively execute market buys and could be left with owning a market order after an "arb" transaction) and sample startup script in the `README`.

**Comment by KentonPrescott** [comment]
Keeper runs, but there's another error related to reading the Oasis API:
```
profit_uniswap_to_oasis = self.oasis_order_size(uniswap_arb_amount) - self.entry_amount
TypeError: unsupported operand type(s) for -: 'NoneType' and 'Wad'
```

I'll let another PR handle the troubleshooting and subsequent patch.

**Pull request merged**